### PR TITLE
chore: release brew bottles and formula from OSS repository

### PR DESCRIPTION
**Pull request by gregmagolan**
# Description

This moves the OSS brew formula & bottle creation to the OSS repository so the brew formula uses artifacts built from the official OSS release and instead of from the same code built in silo pre-copybara.

Not much in the way of new code here. Mostly copy-pasta from silo and it will be removed from there since silo doesn't need to do brew bottles & formula releases for the CLI.

# Type of change

- [x] Chore (any other change that doesn't affect source or test files)

# Test plan

How has this been tested?

- [x] New test cases added
- [x] Manual testing

Ran the CI release step manually and checked results:

```
bazel --bazelrc=.github/workflows/ci.bazelrc --bazelrc=.bazelrc run --config=release //release -- /tmp/aspect/release
```

```
$ ls -la /tmp/aspect/release
total 561136
drwxr-xr-x  28 greg  wheel       896  7 Apr 12:11 .
drwxr-xr-x   3 greg  wheel        96  7 Apr 12:11 ..
-r-xr-xr-x   1 greg  wheel   8145101  7 Apr 12:11 aspect-5.2.1-36-g05323c3.arm64_big_sur.bottle.tar.gz
-r-xr-xr-x   1 greg  wheel   8145102  7 Apr 12:11 aspect-5.2.1-36-g05323c3.arm64_monterey.bottle.tar.gz
-r-xr-xr-x   1 greg  wheel   8513361  7 Apr 12:11 aspect-5.2.1-36-g05323c3.big_sur.bottle.tar.gz
-r-xr-xr-x   1 greg  wheel   8513362  7 Apr 12:11 aspect-5.2.1-36-g05323c3.monterey.bottle.tar.gz
-r-xr-xr-x   1 greg  wheel   8208567  7 Apr 12:11 aspect-5.2.1-36-g05323c3.x86_64_linux.bottle.tar.gz
-r-xr-xr-x   1 greg  wheel  24111136  7 Apr 12:11 aspect-darwin_amd64
-r-xr-xr-x   1 greg  wheel        86  7 Apr 12:11 aspect-darwin_amd64.sha256
-r-xr-xr-x   1 greg  wheel  23844146  7 Apr 12:11 aspect-darwin_arm64
-r-xr-xr-x   1 greg  wheel        86  7 Apr 12:11 aspect-darwin_arm64.sha256
-r-xr-xr-x   1 greg  wheel  22159360  7 Apr 12:11 aspect-linux_amd64
-r-xr-xr-x   1 greg  wheel        85  7 Apr 12:11 aspect-linux_amd64.sha256
-r-xr-xr-x   1 greg  wheel  21102592  7 Apr 12:11 aspect-linux_arm64
-r-xr-xr-x   1 greg  wheel        85  7 Apr 12:11 aspect-linux_arm64.sha256
-r-xr-xr-x   1 greg  wheel  31636992  7 Apr 12:11 aspect-windows_amd64.exe
-r-xr-xr-x   1 greg  wheel        91  7 Apr 12:11 aspect-windows_amd64.exe.sha256
-r-xr-xr-x   1 greg  wheel      1074  7 Apr 12:11 aspect_brew_artifacts.rb
-r-xr-xr-x   1 greg  wheel  23844146  7 Apr 12:11 bazel-5.2.1-36-g05323c3-darwin-arm64
-r-xr-xr-x   1 greg  wheel       103  7 Apr 12:11 bazel-5.2.1-36-g05323c3-darwin-arm64.sha256
-r-xr-xr-x   1 greg  wheel  24111136  7 Apr 12:11 bazel-5.2.1-36-g05323c3-darwin-x86_64
-r-xr-xr-x   1 greg  wheel       104  7 Apr 12:11 bazel-5.2.1-36-g05323c3-darwin-x86_64.sha256
-r-xr-xr-x   1 greg  wheel  21102592  7 Apr 12:11 bazel-5.2.1-36-g05323c3-linux-arm64
-r-xr-xr-x   1 greg  wheel       102  7 Apr 12:11 bazel-5.2.1-36-g05323c3-linux-arm64.sha256
-r-xr-xr-x   1 greg  wheel  22159360  7 Apr 12:11 bazel-5.2.1-36-g05323c3-linux-x86_64
-r-xr-xr-x   1 greg  wheel       103  7 Apr 12:11 bazel-5.2.1-36-g05323c3-linux-x86_64.sha256
-r-xr-xr-x   1 greg  wheel  31636992  7 Apr 12:11 bazel-5.2.1-36-g05323c3-windows-x86_64.exe
-r-xr-xr-x   1 greg  wheel       109  7 Apr 12:11 bazel-5.2.1-36-g05323c3-windows-x86_64.exe.sha256
```

### CODE REVIEW

File: .github/workflows/default.yml [hunk 1]
```diff
@@ -13,7 +13,11 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+        with:
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
```
Comment (alexeagle) [inline .github/workflows/default.yml:18]: why do we need this? I've never seen it before

File: patches/com_grail_bazel_toolchain.patch [hunk 2]
```diff
@@ -0,0 +1,38 @@
+diff --git a/toolchain/cc_toolchain_config.bzl b/toolchain/cc_toolchain_config.bzl
```
Comment (alexeagle) [inline patches/com_grail_bazel_toolchain.patch:1]: I know this patch is just getting copied around, but it really ought to have an upstream PR on it so we could drop in the future. @f0rmiga have you already raised the issue?

File: platforms/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    name = "linux_x86_64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+platform(
+    name = "linux_x86_64_remote",
```
Comment (alexeagle) [inline platforms/BUILD.bazel:38]: this one is probably unused, we don't have RBE in this repo?

File: release/bazel
Comment (alexeagle) [inline release/bazel:1]: is this file load-bearing?

**Review by alexeagle — APPROVED** [review]
cool, feels like we wish we had a place to share this homebrew rule

File: .github/workflows/default.yml (hunk 1 above)
Comment (gregmagolan) [reply .github/workflows/default.yml:18]: I'll add a comment as to why. One of the tests brought in for the brew rules depends on the workspace status command which depends on tags but the default checkout step uses a depth of 1 and doesn't fetch any tags which fails the test

File: release/bazel
Comment (gregmagolan) [reply release/bazel:1]: It is used by the `cli_brew_artifacts` rule for `additional_bins`:
```
cli_brew_artifacts(
    name = "aspect_brew_artifacts",
    additional_bins = [":bazel"],
```

File: platforms/BUILD.bazel (hunk 3 above)
Comment (gregmagolan) [reply platforms/BUILD.bazel:38]: removing

**Pull request merged**